packages/compiler/config/config-loader.ts

```diff
@@ -17,42 +18,53 @@ export const defaultConfig = deepFreeze({
 
 /**
  * Look for the project root by looking up until a `tspconfig.yaml` is found.
- * @param path Path to start looking
+ * @param path Path to the file or the folder to start looking
  */
 export async function findTypeSpecConfigPath(
   host: CompilerHost,
   path: string
 ): Promise<string | undefined> {
-  let current = path;
-  while (true) {
-    let pkgPath = await searchConfigFile(host, current, TypeSpecConfigFilename);
-    if (pkgPath === undefined) {
-      pkgPath = await searchConfigFile(host, current, OldTypeSpecConfigFilename);
-    }
-    // if found either file in current folder, return it
-    if (pkgPath !== undefined) {
-      return pkgPath;
-    }
-    const parent = getDirectoryPath(current);
-    if (parent === current) {
-      return undefined;
+  try {
+    const stats = fs.statSync(path);
+    if (stats.isFile()) {
+      return path;
```

Comment:
here is where we probably need to set the config directory if they provided a file path

Reply:
I know, for example, that we have path interpolation in config based on the config directory, so we likely need to know both the directory and the file name

Reply:
Is there a reason to allow passing in a config directory at all?  It seems like behavior would be simplified if we required them to pass in a config file and did not allow passing in a directory.  This would mean that the 'config search'  would only occur if the path to the config file was not passed in. It would also mean setting the config directory based on the directory containing the config file setting.

Reply:
@markcowl perhaps this warrants some time in a design meeting to iron out the particulars of what we want the behavior to be?

Reply:
We settled on providing an error if a `--config` value is provided and the config is not found, but preserving no error if `--config` is not provided.